sycl/include/sycl/ext/oneapi/experimental/prefetch.hpp

```diff
@@ -71,7 +71,10 @@ inline constexpr bool check_prefetch_acc_mode =
 template <typename T, typename Properties>
 void prefetch_impl(T *ptr, size_t bytes, Properties properties) {
 #ifdef __SYCL_DEVICE_ONLY__
-  auto *ptrGlobalAS = __SYCL_GenericCastToPtrExplicit_ToGlobal<const char>(ptr);
+  auto *ptrGlobalAS =
+      reinterpret_cast<__attribute__((opencl_global)) const char *>(
+          detail::static_address_cast<access::address_space::global_space>(
```

Comment:
See https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_prefetch.asciidoc, it states (for non-accessor overloads):
> Preconditions: ptr must point to an object in global memory.